src/update_agent/actor.rs

```diff
@@ -91,31 +139,50 @@ impl Handler<RefreshTick> for UpdateAgent {
                     "scheduling next agent refresh in {} seconds",
                     pause.as_secs()
                 );
-                Self::tick_later(ctx, pause);
+                actor.tick_later(ctx, pause);
             } else {
                 let update_timestamp = chrono::Utc::now();
                 actor.state_changed = update_timestamp;
-                Self::tick_now(ctx);
+                actor.tick_now(ctx);
             }
             actix::fut::ready(())
         });
 
         // Process state machine refresh ticks sequentially.
         ctx.wait(update_machine);
 
-        Box::pin(actix::fut::ok(()))
+        Box::pin(actix::fut::ok(self.state.clone()))
     }
 }
 
 impl UpdateAgent {
+    /// Cancel the scheduled refresh tick whose handle is stored in the update agent's
+    /// `tick_later_handle` field, if any.
+    fn cancel_scheduled_ticks(&mut self, ctx: &mut Context<Self>) {
+        if let Some(handle) = self.tick_later_handle {
+            ctx.cancel_future(handle);
+            self.tick_later_handle = None;
+        }
+    }
+
     /// Schedule an immediate refresh of the state machine.
-    pub fn tick_now(ctx: &mut Context<Self>) {
-        ctx.notify(RefreshTick {})
+    pub fn tick_now(&mut self, ctx: &mut Context<Self>) {
+        // Cancel scheduled ticks, if any.
+        self.cancel_scheduled_ticks(ctx);
```

Comment:
I think here it is possible to have a race where a self tick msg and a command msg is sent at the same time, and the command msg arrives first.
If this happens, we can't really cancel the scheduled tick like this because the message had alrady been sent. Though I don't think it is too big of a deal to have an extra tick; in practice, this results in two continuous ticks without any delay in between when Zincati is in a steady state (NoNewUpdates and UpdateStaged) where it periodically sends itself refresh ticks.